WS2012R2/lisa/setupscripts/FCOPY_non_ascii.ps1

```diff
@@ -0,0 +1,257 @@
+
+# Checking the input arguments
+if (-not $vmName) {
+    "Error: VM name is null!"
+    return $retVal
+}
+
+if (-not $hvServer) {
+    "Error: hvServer is null!"
+    return $retVal
+}
+
+if (-not $testParams) {
+    "Error: No testParams provided!"
+    "This script requires the test case ID and VM details as the test parameters."
+    return $retVal
+}
+
+#
+# Checking the mandatory testParams. New parameters must be validated here.
+#
+$params = $testParams.Split(";")
+foreach ($p in $params) {
+    $fields = $p.Split("=")
+    
+    if ($fields[0].Trim() -eq "TC_COVERED") {
+        $TC_COVERED = $fields[1].Trim()
+    }
+	if ($fields[0].Trim() -eq "rootDir") {
+        $rootDir = $fields[1].Trim()
+    }
+	if ($fields[0].Trim() -eq "ipv4") {
+		$IPv4 = $fields[1].Trim()
+    }
+	if ($fields[0].Trim() -eq "sshkey") {
+        $sshkey = $fields[1].Trim()
+    }
+}
+
+#
+# Change the working directory for the log files
+# Delete any previous summary.log file, then create a new one
+#
+if (-not (Test-Path $rootDir)) {
+    "Error: The directory `"${rootDir}`" does not exist"
+    return $retVal
+}
+cd $rootDir
+
+# Delete any previous summary.log file, then create a new one
+$summaryLog = "${vmName}_summary.log"
+del $summaryLog -ErrorAction SilentlyContinue
+Write-Output "This script covers test case: ${TC_COVERED}" | Tee-Object -Append -file $summaryLog
+
+$retVal = $True
+
+#
+# Verify if the Guest services are enabled for this VM
+#
+$gsi = Get-VMIntegrationService -vmName $vmName -ComputerName $hvServer -Name "Guest Service Interface"
+if (-not $gsi) {
+    "Error: Unable to retrieve Integration Service status from VM '${vmName}'" | Tee-Object -Append -file $summaryLog
+    return $False
+}
+
+if (-not $gsi.Enabled) {
+    "Warning: The Guest services are not enabled for VM '${vmName}'" | Tee-Object -Append -file $summaryLog
+	if ((Get-VM -ComputerName $hvServer -Name $vmName).State -ne "Off") {
+		Stop-VM -ComputerName $hvServer -Name $vmName -Force -Confirm:$false
+	}
+
+	# Waiting until the VM is off
+	while ((Get-VM -ComputerName $hvServer -Name $vmName).State -ne "Off") {
+		Start-Sleep -Seconds 5
+	}
+	
+	Enable-VMIntegrationService -Name "Guest Service Interface" -vmName $vmName -ComputerName $hvServer 
+	Start-VM -Name $vmName -ComputerName $hvServer
+
+	# Waiting for the VM to run again and respond to SSH - port 22
+	do {
+		sleep 5
+	} until (Test-NetConnection $IPv4 -Port 22 -WarningAction SilentlyContinue | ? { $_.TcpTestSucceeded } )
+}
+
+if ($gsi.OperationalStatus -ne "OK") {
+    "Error: The Guest services are not working properly for VM '${vmName}'!" | Tee-Object -Append -file $summaryLog
+	$retVal = $False
+}
+else {
+	# Define the file-name to use with the current time-stamp
+    $MyDir= "$pwd\"
+	$testfile = "testfile-$(get-date -uformat '%H-%M-%S-%Y-%m-%d').file" 
+    $pathToFile="$MyDir"+"$testfile" 
+
+	# Create a ~100MB sample file with non-ascii characters
```

Comment:
File should be generated only after it's confirmed that the fcopy daemon is present/running first.